components/config-mgmt-service/integration_test/nodes_test.go

```diff
@@ -139,6 +140,373 @@ func TestNodesWithSorting(t *testing.T) {
+						OrganizationName: "org1",
+					},
+					Projects: []string{"two"},
+					Exists:   true,
+				},
+			},
+			ctx:      contextWithProjects([]string{authzConstants.UnassignedProjectID, "two"}),
+			request:  request.Nodes{},
+			expected: []string{"1", "2"},
+		},
+		{
+			description: "Match all projects with status filter",
+			nodes: []iBackend.Node{
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName:         "1",
+						EntityUuid:       newUUID(),
+						Status:           "failure",
+						Platform:         "centos",
+						Environment:      "prod",
+						OrganizationName: "org1",
+					},
+					Projects: []string{},
+					Exists:   true,
+				},
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName:         "2",
+						EntityUuid:       newUUID(),
+						Status:           "success",
+						Platform:         "centos",
+						Environment:      "prod",
+						OrganizationName: "org1",
+					},
+					Projects: []string{"two"},
+					Exists:   true,
+				},
+			},
+			ctx: contextWithProjects([]string{authzConstants.AllProjectsExternalID}),
+			request: request.Nodes{
+				Filter: []string{"status:success"},
+			},
+			expected: []string{"2"},
+		},
+		{
+			description: "Match 'one' project with status filter",
+			nodes: []iBackend.Node{
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName:         "1",
+						EntityUuid:       newUUID(),
+						Status:           "failure",
+						Platform:         "centos",
+						Environment:      "prod",
+						OrganizationName: "org1",
+					},
+					Projects: []string{"one"},
+					Exists:   true,
+				},
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName:         "2",
+						EntityUuid:       newUUID(),
+						Status:           "success",
+						Platform:         "centos",
+						Environment:      "prod",
+						OrganizationName: "org1",
+					},
+					Projects: []string{"two"},
+					Exists:   true,
+				},
+			},
+			ctx: contextWithProjects([]string{"one"}),
+			request: request.Nodes{
+				Filter: []string{"status:success"},
+			},
+			expected: []string{},
+		},
+		{
+			description: "Match unassigned projects with status filter",
+			nodes: []iBackend.Node{
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName:         "1",
+						EntityUuid:       newUUID(),
+						Status:           "failure",
+						Platform:         "centos",
+						Environment:      "prod",
+						OrganizationName: "org1",
+					},
+					Projects: []string{},
+					Exists:   true,
+				},
+				{
+					NodeInfo: iBackend.NodeInfo{
+						NodeName:         "2",
+						EntityUuid:       newUUID(),
+						Status:           "success",
+						Platform:         "centos",
+						Environment:      "prod",
+						OrganizationName: "org1",
+					},
+					Projects: []string{},
+					Exists:   true,
+				},
+			},
+			ctx: contextWithProjects([]string{authzConstants.UnassignedProjectID}),
+			request: request.Nodes{
+				Filter: []string{"status:success"},
+			},
+			expected: []string{"2"},
+		},
+	}
```

Comment:
There's some repetition in these test cases that might make sense to factor out, for example there's a contextWithProjects call in each one, and a lot of redundant info in the nodes; I'm pretty unfamiliar with this stuff and I had to read a few of the test cases to realize that the name and projects were the only significant fields.

Reply:
I will pull more of the data out of the cases and place it in the run section.

Reply:
@phiggins This commit cleaned up the tests https://github.com/chef/automate/pull/222/commits/103dee64149ee2432de04ae87327e9cd677fbf47